src/main/java/tsundere/task/TaskList.java

```diff
@@ -0,0 +1,215 @@
+package tsundere.task;
+
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.time.LocalDate;
+
+import java.util.ArrayList;
+
+import tsundere.ui.Parser;
+import tsundere.exception.GeneralException;
+
+public class TaskList {
+    public static ArrayList<Task> taskList = new ArrayList<>();
+
+    /**
+     * Sets selected Task's status to undone.
+     *
+     * @throws GeneralException If TaskList is empty.
+     */
+    public static String unmark() throws GeneralException {
+
+        if (TaskList.taskList.isEmpty()) throw new GeneralException("What you tryna unmark huh?");
+        try {
+            Task t = TaskList.taskList.get(Integer.parseInt(Parser.name.substring(7, 8)) - 1);
+            if (t.getStatusIcon().equals(" ")) {
+                return("You haven't even started this task dummy!");
+            } else {
+                t.unMark();
+                return(t.toString());
+            }
+        } catch (NumberFormatException e) {
+            throw new GeneralException("Can't you spell?");
+        } catch (IndexOutOfBoundsException e ) {
+            throw new GeneralException("Im pretty sure that's the wrong task number! Check again!");
+        }
+
+    }
+
+    /**
+     * Sets selected Task's status to done.
+     *
+     * @throws GeneralException If TaskList is empty.
+     */
+    public static String mark() throws GeneralException {
+
+        if (TaskList.taskList.isEmpty()) throw new GeneralException("What you tryna mark huh?");
+        try {
+            Task t = TaskList.taskList.get(Integer.parseInt(Parser.name.substring(5, 6)) - 1);
+            if (t.getStatusIcon().equals("X")) {
+                return("You already finished this!");
+            } else {
+                t.markAsDone();
+                return(t.toString());
+            }
+        } catch (NumberFormatException e) {
+            throw new GeneralException("Can't you spell?");
+        } catch (IndexOutOfBoundsException e ) {
+            throw new GeneralException("Im pretty sure that's the wrong task number! Check again!");
+        }
+
+    }
+
+    /**
+     * Removes selected Task from TaskList.
+     *
+     * @throws GeneralException If TaskList is empty.
+     */
+    public static String delete() throws GeneralException {
+
+        if (TaskList.taskList.isEmpty()) throw new GeneralException("What you tryna delete huh?");
+
+        try {
+            int idx = Integer.parseInt(Parser.name.substring(7, 8)) - 1;
+            Task t = TaskList.taskList.get(idx);
+            TaskList.taskList.remove(idx);
+            return getListSize("deleted", t);
+        } catch (NumberFormatException e) {
+            throw new GeneralException("Can't you spell?");
```

Comment:
Consider changing this error message to be a more descriptive one, since the user might have entered a negative integer which is technically not spelling.